### ISSUE TITLE
A possible workaround for games with full-screen issues

### ISSUE THREAD

**Issue by b1r63r** [wine-compholio/wine-staging#226]
My children have been running Spore under Wine the last few days. I managed to get both the Origin installer and Spore (installed through the Origin thingy) running. But we ran into problems with full screen mode on Fedora 21 (gnome 3). The display would be shifted up and to the left, so some of the Spore screen was off screen and some of the desktop showed on the right hand (and part of the notification bar at the bottom). Exact position varied. I see a lot of posts with similar problems in other games.

We then discovered that it happened far less (almost never) for one of the children, so I observed to see what she did differently. And she almost always runs minecraft in full screen first. Not as a wine program. Minecraft runs directly in OpenJDK on fedora. So her brother (who always had problems with Spore) started minecraft in full screen, quit it again and then started Spore. And it works! Now Spore runs correctly for him as well.

We are planning to test with other applications in full screen after a reboot (or just log off and on) to see if any application in full screen before starting wine helps. But I wasn't allowed to do those tests yesterday (when Spore finally worked correctly I wasn't allowed near the computer again).

Perhaps wine lacks something in graphics initialization?

Spore in Wine runs *much* faster than in VirtualBox.

**Comment by Unixware** [comment]
I have the similar issue, fedora 21 KDE.
Fullscreen under wine works fine on any other distro (that uses the usual wine builds) but not on fedora 21. I do remember that wine worked well under Fedora 20. Seems its an issue with your build.

**Comment by slackner** [comment]
@Unixware: I am not aware of any modifications in our builds which influence the fullscreen behaviour. I also have never encountered any issues with fullscreen myself so far, but I am not using KDE. Do you have the 32-bit version of libxcomposite installed?

If the issue remains, could you please test with an unpatched version of wine, for example build from source?

**Comment by Unixware** [comment]
What I found now is that running the wine via a terminal, instead of double-clicking the .EXE,  shows the fullscreen without any issues! So I think something is wrong with the F21.
I have to switch to another distro today, as I noticed there are some issues when flash is playing on the web browser, the display is distorted!

**Comment by Unixware** [comment]
Update: wine-staging works fine on Kubuntu, so I think now, its a bug with Fedora 21.

**Comment by deri82** [comment]
I dont have these losing focus issues anymore, my issues werent in fullscreen applications.

**Comment by slackner** [comment]
Hi everyone,

About one year ago, we announced that we would like to move to more advanced bug tracker. We are sorry for the inconvenience, but now its time to say finally goodbye to the old one. This means, all remaining opened bugs (including this one!) will be closed, and no longer available. If you still need any information from the old bugtracker, please write a mail to webmaster@fds-team.de.

More important, if your issue still exists, please open a new bug report at one of the places mentioned below. For bug reports, please attach terminal output from a recent Wine version (preferably >= 1.8). Please note that in the meantime, Wine Staging has become an official part of WineHQ, so if your issue is already tracked at winehq.org, there is NO NEED to open a new bug report for it. Feel free to add the Wine Staging maintainers (sebastian@fds-team.de, michael@fds-team.de) as CC if you want us to take a look at your bug report.

**NEW BUG TRACKERS:**

* For all Wine AND Wine Staging bugs. USE THIS IF IN DOUBT. (*)
  https://bugs.winehq.org/enter_bug.cgi?product=Wine

* For packaging issues:
  https://bugs.winehq.org/enter_bug.cgi?product=Packaging

* For patch submission:
  https://bugs.wine-staging.com/enter_bug.cgi

Best regards,
Sebastian

----

(*) The "Wine Staging" product in the winehq bugtracker should only be used, when your problem turns out to be a regression introduced in "Wine Staging". Since most problems also occur with Wine, it is recommended to open your bug in the "Wine" product, unless someone can show that it doesn't occur with the development branch.
